### DISCUSSION TITLE
What does the writer return?

### DISCUSSION THREAD

**Discussion by yuki-kimoto** [Ovid/Cor#78] [Q&A] [answered]
What does the writer return?

https://github.com/Ovid/Cor/blob/master/rfc/attributes.md#6233-writeroptional_identifier

```
class SomeClass {
    field $id            :writer;
}

my $thing = SomeClass->new(...);
my $ret = $thing->set_id($new_id);
```

1. The setting value
2. the original value
3. the invocant
4. void

**Comment by leonerd** [comment, marked as answer]
In Object::Pad, the invocant:

https://metacpan.org/pod/Object::Pad#:writer,-:writer(NAME)
https://metacpan.org/release/PEVANS/Object-Pad-0.68/source/t/22fields-accesssors.t#L71

I plan to do the same thing in `use feature 'class'`

**Reply by yuki-kimoto** [reply]
> the invocant

Thanks.
